software/esp-firmware/esp-firmware.ino

```diff
@@ -198,11 +198,12 @@ void loop() {
       break;
     
     case STATE_LOWPOWER:
-      setLEDvalues("0/0", "0/0", "1/250");
+      //no need to set LED state, it's already done
+      //setLEDvalues("0/0", "0/0", "1/250");
       // shut down ESP if low on battery
-      if(lowBattery || emptyBattery)
+      if(lowBattery || emptyBattery || millis() > stateTimeout)
```

Comment:
I like your idea of an inactivity timeout... however, this line will break another path:

When turning the wiFred off, first the software goes into STATE_LOWPOWER (after 5 seconds in STATE_LOCOS_OFF) and then waits there for either the power to turn off (RC delay at enable pin of the voltage regulator) or the user to turn the wiFred on again (that's the if(!allLocosInactive()) below). It intentionally goes into deepSleep only if the battery is low/empty.

Your change here will break in the following sequence

1) turn off the wiFred
2) wait for the LEDs to turn off
3) try turning it on again immediately

Maybe just add another global bool "inactivityTimeout" just like "emptyBattery"?  Or put that inside locoHandling.cpp and implement a getter, if you prefer a slightly more OOP approach.

Heiko

Reply:
Well, I thinks it a rare situation to do so - but may easily be addressed by increasing the "state-timeout" for "STATE_LOWPOWER" to above RC-timer. E.g from 500 to 5000
Line 192:    +    switchState(STATE_LOWPOWER, 5000);

Reply:
In that case, it goes immediately  to sleep on low batt, but waits if entering STATE_LOWPOWER if reason is either turn-off, or cased by activity timer. In case of turn-off, user has RC-timer timeout to turn back on, in case activity timer, it just waits antoter 5 sec :)